llvm/include/llvm/DebugInfo/Symbolize/SymbolizableObjectFile.h

```diff
@@ -83,8 +83,30 @@ class SymbolizableObjectFile : public SymbolizableModule {
     // getNameFromSymbolTable.
     uint32_t ELFLocalSymIdx;
 
+    // Both are false if this is not a XCOFF local symbol.
+    bool IsXCOFFSDSymbol;
+    bool IsXCOFFLDSymbol;
```

Comment:
if I understand correct, you want to priority the symbols which has the same address, can you do something like  https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/MC/MCDisassembler/MCDisassembler.h#L28 
and 
https://github.com/llvm/llvm-project/blob/main/llvm/lib/MC/MCDisassembler/MCDisassembler.cpp#L85

it will le the Symbolize has the same priority behaviors' as llvm-objdump for xcoff symbols ?

I do not think in XCOFF , when priority the same address symbols, we care about the size of the symbol,  we care about the  `symbol type` and `storage mapping class`.

Reply:
> Is it possible for these to both be true? If not, perhaps an enum, something like XCOFFSymbolClass, would be more appropriate? In fact, an enum might be better in general, to avoid having to keep track of multiple different fields.

Thanks @jh7370. This is a good idea. Updated.

Reply:
```
I do not think we care about the size of the symbol when priority the same address symbols in XCOFF , we care about the symbol type and storage mapping class.
```

Thanks @diggerlin for pointing out these existing functions. The implementation here is same with those places. Except that here we only care about the symbol type. Because the symbols in llvm-symbolizer are already filtered. See the function `SymbolizableObjectFile::addSymbol()`. Using address and symbol type are enough to sort the filtered symbols.

However, I agree with you that it is strange we still need to sort based on symbol size. We need to prioritize based on size for below two symbols(actually I think this is an existing XCOFF symbol table bug) :
```
[1]     m   0x00000000     .text     1  unamex                    **No Symbol**
[2]     a4  0x00000000       0    0     SD       PR    0    0
[3]     m   0x00000000     .text     1  extern                    .foo
[4]     a4  0x00000019       0    0     SD       PR    0    0
```
( Compiled from `llvm/test/tools/llvm-symbolizer/xcoff-sd-symbol.ll` in this patch, `llc -filetype=obj xcoff-sd-symbol.ll -mtriple=powerpc-aix-ibm-xcoff -function-sections`)

If there is no label under symbol 1(.text csect for all no csect functions), we can remove the symbol 1 and symbol 2, right? And with this redundant symbol, there is not way to prioritize symbol 1 and symbol 3. Sizes are a workaround better than names?

Reply:
agree,  compare size also look work in following scenario.  we put var[TC] high priority than TC0.

[19]    m   0x00000048     .data     1  unamex                    TOC
[20]    a4  0x00000000       0    0     SD      TC0    0    0
[21]    m   0x00000048     .data     1  unamex                    var
[22]    a4  0x00000004       0    0     SD       TC    0    0